### DISCUSSION TITLE
Is it possible to  modify session expiry time at the time of login flow by replacing existing default time?

### DISCUSSION THREAD

**Discussion by shreeharsha-factly** [ory/kratos#3118] [Q&A] [answered]
For the user, the default session would be 24 hours but I want it to be customizable at the login flow. So the user can opt for a different session time like staying signed in for a week at the time of log in. Can we do that?

**Comment by Benehiko** [comment, marked as answer]
Hi @shreeharsha-factly 

There is no user-facing way for users to extend the session lifespan. What is the use case for extending a session per user?

What if you set the session lifespan to a week and then force a re-authentication on any "dangerous" tasks. For example when the user does a payment or views sensitive information.
https://www.ory.sh/docs/kratos/session-management/refresh-extend-sessions#forcing-session-refresh

As an administrator you could force the session to extend, but then you would need to save the settings per user, get all their sessions on your server in a cron job and do an extension through the admin API. 
https://www.ory.sh/docs/kratos/session-management/list-revoke-get-sessions#listing-all-sessions-of-a-user
https://www.ory.sh/docs/kratos/session-management/refresh-extend-sessions#refreshing-sessions-as-administrator

**Reply by shreeharsha-factly** [reply]
Thanks, @Benehiko for the reply. We want to implement feature like `remember me` so the user can opt for longer active sessions instead of the default one.

**Reply by Benehiko** [reply]
Hi @shreeharsha-factly 

please see https://github.com/ory/kratos/discussions/1661
